haystack/retriever/dense.py

```diff
@@ -240,16 +240,19 @@ def _get_predictions(self, dicts):
         else:
             disable_tqdm = not self.progress_bar
 
-        for i, batch in enumerate(tqdm(data_loader, desc=f"Creating Embeddings", unit=" Batches", disable=disable_tqdm)):
-            batch = {key: batch[key].to(self.device) for key in batch}
-
-            # get logits
-            with torch.no_grad():
-                query_embeddings, passage_embeddings = self.model.forward(**batch)[0]
-                if query_embeddings is not None:
-                    all_embeddings["query"].append(query_embeddings.cpu().numpy())
-                if passage_embeddings is not None:
-                    all_embeddings["passages"].append(passage_embeddings.cpu().numpy())
+        with tqdm(total=len(data_loader)*self.batch_size, unit=" Docs", desc=f"Create embeddings", position=1,
```

Comment:
Changed this one to also show docs as unit. This seems more intuitive if you want to track the overall progress.

However, one issue is remaining here: Currently the logs get one new line per iteration of this 2nd progressbar:

```
Updating embeddings:   0%|          | 0/48377 [00:00<?, ? Docs/s]
Create embeddings:   0%|          | 0/10048 [00:00<?, ? Docs/s]
Create embeddings:   1%|          | 64/10048 [00:39<1:43:47,  1.60 Docs/s]
Create embeddings:   1%|▏         | 128/10048 [01:20<1:43:52,  1.59 Docs/s]
Create embeddings:   2%|▏         | 192/10048 [01:58<1:41:38,  1.62 Docs/s]
```
This is again polluting the logs, so would be better to update the second bar instead of new lines. Gave it a quick try with tqdm.auto but this did not help. Had no further time to investigate it right now. Maybe you want to give it a try?

Reply:
> I think we need to use `from tqdm.autonotebook import tqdm` instead of `from tqdm.auto import tqdm`.
> https://github.com/tqdm/tqdm/blob/bcce20f771a16cb8e4ac5cc5b2307374a2c0e535/tqdm/autonotebook.py

@lalitpagaria From my understanding `tqdm.auto` is the more general case which includes `tqdm.autonotebook`. 

Tried it quickly and `tqdm.autonotebook` didn't fix it for me. However, I found that the reason was my PyCharm configuration (see https://stackoverflow.com/questions/64727187/tqdm-multiple-progress-bars-with-nested-for-loops-in-pycharm). 
So I think we are good here...